### DISCUSSION TITLE
Ship Systems - Weapons

### DISCUSSION THREAD

**Discussion by alexanderson1993** [Thorium-Sim/thorium-nova#429] [Feature Proposals]
Let's talk about weapons.

### The Vision For Battles

Everything in Thorium Nova is configurable, including the speed and maneuverability of ships and weapons. That means, if you wanted to, you could create a dogfight-y universe where everything is snappy and quick.

For the default universe, though, I want to optimize for more careful, methodical battles that prefer strategy and tactics over raw power. A battle between two evenly matched ships will come down to which side is better at managing their weapons and coordinating their response. Battles will involve nearly every person on the bridge in some way. Though we won't dig into how every other person is involved in battles quite yet.

Also, due to the sheer speeds and distances involved, it will _not be possible to fire weapons at warp speed_. If you want to have some kind of high-speed chase battle, you'll have to do it at impulse speed.

A ships "hit points" consist of its shields (if raised) and hull. Both must be taken to 0% for the target to be destroyed, though any damage has a chance of breaking a ship system.

### Weapons Range and Targeting

The pilot has a visual range of 10,000km, which is just less than the diameter of the Earth. That also seems a reasonable weapons range. They should probably have exactly the same range since they'll need to coordinate movement and weapons fire.

Though perhaps weapons range will have to be expanded to allow for torpedos to have a longer range than phasers.

Objects are targeted by clicking on them on the targeting grid. That's all that needs to be done for that.

### Phasers/Beam Weapons

The primary weapons are beam weapons. These can hit targets virtually instantly within a spherical sector - kind of like an arc, but 3D. Basically a cone, where the angle and width of the cone can be configured. Any ships within this cone can be hit by the beam. Ships are equipped with multiple phaser banks, perhaps with each having a different angle and width.

Beam weapons convert energy on the ship into destructive power, and have to be charged before they can be fired. There will likely be some kind of space magic to amplify the power of the beam weapons, because fusion reactors just don't have enough juice to generate the necessary power for this level of destruction. Anyway, the crew member will charge the phaser bank, which will draw energy away from the rest of the ship.

Once charged and with a target in range, the beam weapon can be fired continuously until it has completely discharged. Phasers will automatically track their target, so long as it is in range and unobstructed. Every tick that the beam weapon is firing will inflict some configurable amount of damage on the target. Phasers will be especially effective against shields, but not as effective against hull.

Beam weapons also produce heat - minute amounts as long as they are charged and larger amounts when they fire. Heat leads to lower efficiency, which increases the amount of power required to have the same effectiveness. Probably that means low efficiency makes beam weapons charge slower, but it could also make beam weapon blasts weaker. Efficiency is increased by maintenance crews using inventory to replace worn out equipment in the phasers.

#### Power Management

Just a quick note: A big part of battles will be managing power on the ship. Phasers, shields, and engines need lots of power to operate, and using one comes at the expense of the other. The players have to decide whether they will take a more defensive stance and optimize for keeping their shields charged vs fighting back with their phasers vs running away with their engines.

### Torpedos/Projectiles

Torpedos are projectile weapons fired from a single direction, usually fore and aft though port and starboard launchers are possible as well. Torpedos have a much larger range and have tracking capabilities, allowing them to follow targets. This means torpedos are not effective at short range. The torpedo rotation speed and velocity will be configurable.

In fact, what I want is for torpedos to be composed to two parts: A casing which determines the rotation and velocity of the torpedo, and a warhead which determines the yield, or damage the torpedo does. Of course, we probably don't want to have super buffed torpedos so maybe that's a bad idea. But it could provide some interesting opportunities.

So, torpedos are limited by inventory. In fact, torpedo warheads _and_ casings are initially configured as inventory items. The ship will have some number of each when the flight starts, and only the torpedos in torpedo launcher rooms will be able to be launched. If they launch them all, they'll have to transfer cargo from another room.

Torpedos will be the inverse of beam weapons - less effective against shields, more effective against hull. Torpedos are also much less accurate. If they don't hit, they don't hit. Maybe they have a limited amount of fuel, and once they run out they can't turn anymore. So it's possible they blow past their target due to evasive maneuvers, but the torpedo comes back around for another go until its out of fuel.

Torpedos also take a long time to prepare and load. Maybe there should be some kind of extra targeting required for torpedos, not just a "set it, fire it, and forget it" to make it so using torpedos is a bit weightier.

Torpedos require a spike in power use to do the initial launch. If there's not enough power, the torpedo will travel at a lower speed.

---

That's all for now. Comments, questions, feedback welcome!

**Comment by 1deejay** [comment]
Being able to target different functions on a ship would be nice. Nova would need to track damage for specific systems on opposing ships as well.

**Reply by alexanderson1993** [reply]
The main* difference between opposing ships and the crew's ship is the `isPlayerShip` flag. Targeting specific systems is the plan.

*To keep simulation computation fast, NPC ships will likely not have sophisticated room or crew simulations. Crew simulations in the player ship are designed to only _buff_ systems, not nerf them, so by default, this will give a handicap to the player ship, which will hopefully make the game easier.

**Reply by 1deejay** [reply]
This makes sense, but as we go along I know that as a level designer, sometimes you will want to incapacitate a ship either to run away, or board and take something from the other ship and being able to target specific systems will make that more feasible.

Then when it comes to ship design, some components can be in better or worse places, I'm not sure how feasible that part of design will be but it should be fun.

Then if it does not have the player tag, you can build into the NPC ship cooldowns for their abilities. 5 second laser fire, 10 second torpedo, etc. Then the values of fire rate, power, breakpoint, etc can be played with when it comes to tuning how different ships feel to fight against.
